Array.cc

```diff
@@ -101,14 +101,24 @@ void Array::_duplicate(const Array &a)
  */
 void Array::update_length(int)
 {
-    int length = 1;
+    unsigned long long  length = 1;
```

Comment:
In most cases 'unsigned long long' will mean the same thing as 'uint64_t,' but there is no guarantee that's the case because it could be larger. The type name 'uint64_t' is always going to be an unsigned 64 bit integer.